quantus/metrics/robustness/relative_representation_stability.py

```diff
@@ -0,0 +1,335 @@
+         - Compute relative representation stability objective, find max value with respect to `xs`
+         - In practise we just use `max` over a finite `xs_batch`
+        """
+        result = super().__call__(
+            model=model,
+            x_batch=x_batch,
+            y_batch=y_batch,
+            explain_func=explain_func,
+            explain_func_kwargs=explain_func_kwargs,
+            a_batch=a_batch,
+            device=device,
+            softmax=softmax,
+            channel_first=channel_first,
+            model_predict_kwargs=model_predict_kwargs,
+            s_batch=None,
+            batch_size=batch_size,
+        )
+        return list(result) if isinstance(result, Iterable) else float(result)  # noqa
+
+    def relative_representation_stability_objective(
+        self,
+        l_x: np.ndarray,
+        l_xs: np.ndarray,
+        e_x: np.ndarray,
+        e_xs: np.ndarray,
+    ) -> np.ndarray:
+        """
+        Computes relative representation stabilities maximization objective
+        as defined here https://arxiv.org/pdf/2203.06877.pdf by the authors.
+
+        Parameters
+        ----------
+        l_x: np.ndarray
+            Internal representation for x_batch.
+        l_xs: np.ndarray
+            Internal representation for xs_batch.
+        e_x: np.ndarray
+            Explanations for x.
+        e_xs: np.ndarray
+            Explanations for xs.
+
+        Returns
+        -------
+        rrs_obj: np.ndarray
+            RRS maximization objective.
+        """
+
+        # fmt: off
+        nominator = (e_x - e_xs) / (e_x + (e_x == 0) * self._eps_min)  # prevent division by 0
+        nominator = np.linalg.norm(np.linalg.norm(nominator, axis=(-1, -2)), axis=-1)  # noqa
+        # fmt: on
+        denominator = l_x - l_xs
+        denominator /= l_x + (l_x == 0) * self._eps_min  # prevent division by 0
+        denominator = np.linalg.norm(denominator, axis=-1)
+        denominator += (denominator == 0) * self._eps_min
+        return nominator / denominator
+
+    def evaluate_batch(
+        self,
+        model: ModelInterface,
+        x_batch: np.ndarray,
+        y_batch: np.ndarray,
+        a_batch: Optional[np.ndarray],
+        *args,
+        **kwargs,
+    ) -> np.ndarray:
+        """
+        Parameters
+        ----------
+        model: tf.keras.Model, torch.nn.Module
+            A torch or tensorflow model that is subject to explanation.
+        x_batch: np.ndarray
+            4D tensor representing batch of input images.
+        y_batch: np.ndarray
+            1D tensor, representing predicted labels for the x_batch.
+        a_batch: np.ndarray, optional
+            4D tensor with pre-computed explanations for the x_batch.
+        args:
+            Unused.
+        kwargs:
+            Unused.
+
+        Returns
+        -------
+        ris: np.ndarray
+            The batched evaluation results.
+
+        """
+        batch_size = x_batch.shape[0]
```

Comment:
While the code is quite self-explanatory, for a new user, it might not be. Please look at any other metrics and add similar comments in the three evaluate_batch implementations for this PR

Reply:
done